contacts/main.js

```diff
@@ -0,0 +1,566 @@
+        rerenderContacts();
+      }
+      break;
+    case 'checkbox-custom':
+      changeDeleteAllButtonValueAndState();
+      break;
+  }
+}
+///////////////////////////////////////////////////////////////////
+
+function changeDeleteAllButtonValueAndState() {
+  let button = document.getElementById('delete-all-button');
+  let numberOfCheckedContacts = 0;
+  numberOfCheckedContacts = getSelectedContactsIDs().length;
+  if (numberOfCheckedContacts != 0) {
+    button.innerHTML = 'Delete All ' + numberOfCheckedContacts.toString();
+    button.disabled = false;
+  } else {
+    button.innerHTML = 'Delete All';
+    button.disabled = true;
+  }
+}
+
+
+function resetDeleteAllButtonValueAndState() {
+  document.getElementById('delete-all-button').innerHTML = 'Delete All';
+  document.getElementById('delete-all-button').disabled = true;
+}
+
+function deletePerson(personId) {
+  for (let i = 0; i < contactsList.length; i++)
+    if (contactsList[i].id === personId) {
+      contactsList.splice(i, 1);
+      break;
+    }
+}
+
+function incrementLikes(personId) {
+  for (let i = 0; i < contactsList.length; i++)
+    if (contactsList[i].id === personId) {
+      if (contactsList[i].likes < 99) {
+        contactsList[i].likes += 1;
+        break;
+      }
+    }
+}
+
+function decrementLikes(personId) {
+  for (let i = 0; i < contactsList.length; i++)
+    if (contactsList[i].id === personId) {
+      if (contactsList[i].likes > 0) {
+        contactsList[i].likes -= 1;
+        break;
+      }
+    }
+}
+
+function addNewPerson() {
+  try {
+    let newPerson = getTheNewPerson();
+    contactsList.push(newPerson);
+    rerenderContacts(contactsList);
+    changeAddButtonState();
+  } catch (err) {
+    alert(err.message);
+  }
+}
+
+function getTheNewPerson() {
+  const id = generateUniqueID();
+  const firstName = getFirstNameInput();
+  const lastName = getLastNameInput();
+  const image = getImageProfileInput();
+  let likes = 0;
+  let obj = {
+    id: id,
+    firstName: firstName,
+    lastName: lastName,
+    image: image,
+    likes: likes
+  };
+  let newPerson = new Person(obj);
+  return newPerson;
+}
+
+
+
+function clearButtonAction() {
+  clearContactsLikes();
+  rerenderContacts();
+  clearFavsListHTML();
+}
+
+function clearContactsLikes() {
+  contactsList.forEach((item) => {
+    item.likes = 0;
+  })
+}
+
+function clearFavsListHTML() {
+  document.getElementById('favs-list').innerHTML = '';
+}
+
+
+function changeAddButtonState() {
+  let first_name = document.getElementById('first-name').value;
+  let last_name = document.getElementById('last-name').value;
+  let image = document.getElementById('profile-image').value;
+  if (first_name != '' && last_name != '' && image != '')
+    document.getElementById('add-button').disabled = false;
+  else
+    document.getElementById('add-button').disabled = true;
+}
+
+
+
+function makeAllConections() {
+  document.getElementById('add-button').onclick = addNewPerson;
```

Comment:
instead of doing this, i would personally add onClick on the button (for all the buttons that you attached events handlers here), because it is more visible when reading the code